### ISSUE TITLE
golint seems unable to recognise constants which are typed-iota values with the comment at the end of the line

### ISSUE THREAD

**Issue by sjmudd** [golang/lint#502]
I noticed this when cleaning up some code in another repo.
This is easy to show in the example in a repo I provided here: https://github.com/sjmudd/golintbug/blob/master/main.go.

I think the code is correct and the comment is provided correctly yet `golint` does not agree with me.
Example output I see is shown [here](https://github.com/sjmudd/golintbug/blob/master/README.md). The other constant settings are all accepted as I would expect.

If you agree then please consider adapting `golint` to recognise this comment and not complain.  If not please point me at how I should comment this correctly.

Thanks (and Happy New Year).

**Comment by sjmudd** [comment]
`golint` is from the repo from today (2020-01-03) and `go` reports `go version go1.15.5 darwin/amd64`.

**Comment by sjmudd** [comment]
I notice other related bugs:
- golang/lint#208
- golang/lint#275

However, it is not clear to me that the trailing comments are inappropriate as `golint` does not complain about the other in-line comments in my sample code. If inline comments were not acceptable I would expect warnings about all the constants yet the only one that generates an issue is the one with the typed `iota` constant.

**Comment by sjmudd** [comment]
To avoid others having to jump between github repos the example line that generates an error is:

```
const (
...
	GoLintBugConst1 GoLintBug = iota          // GoLintBugConst1 constant
...
)
```

The error reported by `golint` is:

```
main.go:10:2: exported const GoLintBugConst1 should have comment (or a comment on this block) or be unexported
```

**Comment by dherbst** [comment]
As far as I know inline comments are not supported by `godoc`, thus `golint` is not considering them.   If you move your comment to the line above the declaration `golint` should stop complaining.  I've noticed that `golint` only reports the first error and does not report all of them in the documentation category.

From https://blog.golang.org/godoc:

> The convention is simple: to document a type, variable, constant, function, or even a package, write a regular comment directly preceding its declaration, with no intervening blank line. Godoc will then present that comment as text alongside the item it documents. For example, this is the documentation for the fmt package's Fprint function...

Does that help?

**Comment by sjmudd** [comment]
Not really.

The full same code I provided had:

```
// GoLintBug is really an int
type GoLintBug int

const (
	intConst1                 = 1             // int constant
	intConst2                 = 2             // int constant
	stringConst               = "stringConst" // string constant
	GoLintBugConst1 GoLintBug = iota          // GoLintBugConst1 constant
	GoLintBugConst2                           // GoLintBugConst2 constant
	GoLintBugConst3                           // GoLintBugConst3 constant
)
```

`golint` only complains about the `GoLintBugConst1` definition and not the others.
So if inline comments are not supported why do I not get more warnings? I'd expect to get 6 warnings and only one.

Hence the _bug_ report.

**Comment by sjmudd** [comment]
While I have closed this I looked at the source code: https://github.com/golang/tools/blob/master/godoc/godoc.go and do not see a way to file an "issue" in github for this repo. So it is not clear to me how to suggest an improvement / change to the godoc behaviour or see if others have suggested this and why it has not been considered acceptable etc.

So not sure how to follow up on this. I agree that golint should follow godoc behaviour, but do also think that documenting inline comments would make source code easier to read.  For now I'll just have to adjust my source code accordingly.
